tests/conftest.py

```diff
@@ -259,3 +269,100 @@ def global_sa_results_dir(datadir: Path) -> Path:
     gsa_data_set[:] = global_sensitivity_analysis
     file.close()
     return result_dir
+
+
+def _create_and_populate_hm_result_file(
+    result_dir: Path,
+    result: np.ndarray,
+    dataset_key: str,
+    limits: Sequence[Tuple[float, float]],
+) -> None:
+    result_dir.mkdir(parents=True, exist_ok=True)
+    result_filepath = result_dir / "result"
+
+    file = h5py.File(result_filepath, "x", libver="latest", locking=False)
+    meta_group = file.create_group(META_GROUP_NAME, track_order=True)
+    data_group = file.create_group(HISTORY_MATCHING_GROUP_NAME, track_order=True)
+
+    dataset = data_group.create_dataset(
+        dataset_key,
+        shape=result.shape,
+        dtype=np.float64,
+        maxshape=tuple(None for _ in result.shape),
+    )
+
+    objective_functions = {
+        "observed_curve_1": {"trend_id": "trend_1", "property_id": "holdup"},
+        "observed_curve_2": {"trend_id": "trend_2", "property_id": "pressure"},
+    }
+
+    fake_meta = {
+        "parametric_var_1": {
+            "parametric_var_id": "parametric_var_1",
+            "parametric_var_name": "mg",
+            "min_value": limits[0][0],
+            "max_value": limits[0][1],
+            "objective_functions": objective_functions,
+            "data_index": 0,
+        },
+        "parametric_var_2": {
+            "parametric_var_id": "parametric_var_2",
+            "parametric_var_name": "mo",
+            "min_value": limits[1][0],
+            "max_value": limits[1][1],
+            "objective_functions": objective_functions,
+            "data_index": 1,
+        },
+    }
+
+    meta_group.attrs[HISTORY_MATCHING_GROUP_NAME] = json.dumps(fake_meta)
+    dataset[:] = result
+
+    file.swmr_mode = True
+    file.close()
+
+
+@pytest.fixture()
+def hm_probabilistic_results_dir(datadir: Path) -> Path:
+    """
+    Crete a History Matching result folder with a populated result file for each type of analysis
+    (probabilistic and deterministic).
+    """
+    import numpy as np
+
+    result_dir = datadir / "main-HM-probabilistic"
+    probabilistic_result = np.array(
+        [[0.1, 0.22, 1.0, 0.8, 0.55], [3.0, 6.0, 5.1, 4.7, 6.3]]
+    )
+    limits = [(0.0, 1.0), (2.5, 7.5)]
+
+    _create_and_populate_hm_result_file(
+        result_dir=result_dir,
+        result=probabilistic_result,
+        dataset_key=HISTORY_MATCHING_PROBABILISTIC_DSET_NAME,
+        limits=limits,
+    )
+
+    return result_dir
+
+
+@pytest.fixture()
+def hm_deterministic_results_dir(datadir: Path) -> Path:
+    """
+    Crete a History Matching result folder with a populated result file for each type of analysis
```

Comment:
```suggestion
    Create a History Matching result folder with a populated result file for each type of analysis
```

But the doc in general is incorrect, it is only creating one type of analysis here.

Reply:
You're correct. I missed this one while refactoring things.